### ISSUE TITLE
All notes deleted

### ISSUE THREAD

**Issue by ITCMD** [nextcloud/notes#713]
All of a sudden, all notes for all users are gone.

No other files or directories were effected.

It says the file cannot be opened. I checked and sure enough there were no files present in that folder. After scanning with occ, now all users have no notes.

What happened?? I'm very frustrated. I've lost all of my school notes and a lot of personal notes.
[nextcloud.log](https://github.com/nextcloud/notes/files/6474286/nextcloud.log)

**Comment by korelstar** [comment]
Ouh, that's very bad. I can understand your frustrations.

Please provide more information about your setup (see issue template: server version, operating system, notes app version, storage configuration, encryption configuration, which clients are synced with your server using which software, ...).

What exactly happened when the problem occurred? Was there a synchronization with a client before? Maybe an error on the storage disk?

**Comment by ITCMD** [comment]
OS was Ubuntu 21.04 server. I had the latest version of nextcloud as of April 12th 2021. I've already replaced it with another system altogether so I cant tell you much more than that.
I figured it was a disk issue at first, but there were no errors listed, smart health was good, and I even ran a full badblock disk scan and found nothing off. 

There were no corrupted files or damaged settings, just the notes folder was completely empty despite having 30+ notes.

I know there's a setting that deletes old files in Nextcloud. I definitely did not turn it on but I checked anyways and can confirm it was off so that's not the issue.

I hadn't used my notes for about a week since I was on vacation so I cant say for sure exactly when it happened. I believe it was the 12th or 13th that the errors started showing up in the log, but dont quote me on that.

**Comment by korelstar** [comment]
According to the log file, the first failed access to a note was at 2021-05-13T16:36:53. At this time, a request was made to `/nextcloud/index.php/apps/notes/notes/29755`. On such a request, the notes app asks the Nextcloud server for the file with id 29755. Then, the notes app tries to read this file (`/var/www/html/nextcloud/data/lucas/files/Notes/Birthday IDEAS.txt` in this case). This failed with message "No such file or directory". That's interesting, because this means that Nextcloud server still thinks that file id 29755 exists and the server still "knows" that file id 29755 is associated with `/var/www/html/nextcloud/data/lucas/files/Notes/Birthday IDEAS.txt`. This means, that the file was not deleted/moved using Nextcloud, but it was deleted/moved/broken directly on the filesystem. The Nextcloud server wasn't aware at this point, that the file has gone from filesystem.

Furthermore, I found many log entries with the request `MOVE /nextcloud/remote.php/dav/uploads/lucas/338b53f2ac5c6a4cb9225345f9953cad/.file` and message `Could not open file`. I'm not sure how to interpret this, but it may means, that there are other problems on your server regarding file access.

In summary: I'm very sad that I can't find the exact source of the problem, but it looks like the notes files where moved/deleted/broken somewhere outside the Nextcloud server. Therefore, I can't see any indicators that the notes app did caused this data loss. Hence, I suggest to restore the files from your backup -- in the best case on another server with new hardware.

**Comment by ITCMD** [comment]
Thanks for analyzing the logs. I knew that it was showing that error but couldn't figure out if there was a cause hidden in there somewhere.

The backup had the files missing too. Not sure what was going on, but I have it set with ftp instead now. Hopefully new os / new system will have less issues 

/shrug
